src/key/helper.js

```diff
@@ -392,7 +398,12 @@ export function isValidEncryptionKeyPacket(keyPacket, signature) {
 }
 
 export function isValidDecryptionKeyPacket(signature, config) {
-  if (config.allowInsecureDecryptionWithSigningKeys) {
+  const allowedToSign = !signature.keyFlags ||
+    (signature.keyFlags[0] & enums.keyFlags.sign) !== 0 ||
+    (signature.keyFlags[0] & enums.keyFlags.certifyKeys) !== 0 ||
+    (signature.keyFlags[0] & enums.keyFlags.authentication) !== 0;
+
+  if (allowedToSign && config.allowInsecureDecryptionWithSigningKeys) {
```

Comment:
This is very nitpicky but can we call this
```suggestion
  const isSigningKey = !signature.keyFlags ||
    (signature.keyFlags[0] & enums.keyFlags.sign) !== 0 ||
    (signature.keyFlags[0] & enums.keyFlags.certifyKeys) !== 0 ||
    (signature.keyFlags[0] & enums.keyFlags.authentication) !== 0;

  if (isSigningKey && config.allowInsecureDecryptionWithSigningKeys) {
```
(as normally the determination of whether we can sign with it is slightly different)